hash.go

```diff
@@ -230,6 +230,31 @@ func (h *evpHash) sum(out []byte) {
 	runtime.KeepAlive(h)
 }
 
+func (h *evpHash) clone() (*evpHash, error) {
```

Comment:
Maybe a comment here on the depth of the clone?  Is it shallow or deep? Likewise for other clones below

Reply:
Make sense. I'll document that it is a deep clone.